src/utils/scales-utils.js

```diff
@@ -656,8 +659,11 @@ export function getMissingScaleProps(props, data, attributes) {
  * Return a d3 scale that returns the literal value that was given to it
  * @returns {function} literal scale.
  */
-export function literalScale() {
+export function literalScale(defaultValue) {
   function scale(d) {
+    if (d === undefined) {
+      return defaultValue;
```

Comment:
what if default value is undefined? it should still return undefined right? Add a test case?